File: app/src/main/java/id/niteroomcreation/archcomponent/feature/detail/DetailViewModel.java
package id.niteroomcreation.archcomponent.feature.detail;

import androidx.lifecycle.LiveData;

import id.niteroomcreation.archcomponent.base.BaseViewModel;
import id.niteroomcreation.archcomponent.domain.data.local.entity.FavouriteEntity;
import id.niteroomcreation.archcomponent.domain.data.local.entity.MovieEntity;
import id.niteroomcreation.archcomponent.domain.data.local.entity.TvShowEntity;
import id.niteroomcreation.archcomponent.domain.repositories.RepositoryImpl;

/**
 * Created by Septian Adi Wijaya on 09/05/2021.
 * please be sure to add credential if you use people's code
 */
public class DetailViewModel extends BaseViewModel {

    private int objId;
    private final RepositoryImpl repository;

    public DetailViewModel(RepositoryImpl repository) {
        this.repository = repository;
    }

    public void setSelectedEntity(int objId) {
        this.objId = objId;
    }

    public LiveData<MovieEntity> getMovieById() {
        return repository.getMovieById(objId);
    }

    public LiveData<TvShowEntity> getTvShowById() {
        return repository.getTvShowById(objId);
    }

    public void updateMoviesOnFav(MovieEntity entity) {
        repository.updateMovie(entity);
    }

    public void updateTvShowOnFav(TvShowEntity entity) {
        repository.updateTvShow(entity);
    }

    public void insertIntoFav(FavouriteEntity entity) {
        repository.insertIntoFav(entity);
    }

    public void deleteFav(FavouriteEntity entity) {
        repository.deleteFav(entity);
    }
}

File: app/src/main/java/id/niteroomcreation/archcomponent/feature/detail/DetailViewModel.kt
package id.niteroomcreation.archcomponent.feature.detail

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.viewModelScope
import id.niteroomcreation.archcomponent.base.BaseViewModel
import id.niteroomcreation.archcomponent.domain.data.remote.response.BaseResponse
import id.niteroomcreation.archcomponent.domain.data.remote.response.movies.by_id.MoviesById
import id.niteroomcreation.archcomponent.domain.data.remote.response.movies.reviews.MovieReviews
import id.niteroomcreation.archcomponent.domain.data.remote.utils.ApiResponse
import id.niteroomcreation.archcomponent.domain.repositories.RepositoryImpl
import kotlinx.coroutines.launch

/**
 * Created by Septian Adi Wijaya on 09/05/2021.
 * please be sure to add credential if you use people's code
 */
class DetailViewModel(private val repository: RepositoryImpl) : BaseViewModel() {

    private var respondResult_ = MutableLiveData<ApiResponse<MoviesById>>()
    val respondResult = respondResult_

    private var respondResultReview_ = MutableLiveData<ApiResponse<BaseResponse<MovieReviews>>>()
    val respondResultReview = respondResultReview_

    fun getDetail(id: Int) {
        viewModelScope.launch {
            respondResult_.postValue(repository.getMovieById(id))
        }
    }

    fun getMovieReview(id: Int) {
        viewModelScope.launch {
            respondResultReview_.postValue(repository.getReviewByMovie(id))
        }
    }

}